xknx/io/knxip_interface.py

```diff
@@ -139,24 +139,34 @@ async def _start(self) -> None:
 
     async def _start_automatic(self) -> None:
         """Start GatewayScanner and connect to the found device."""
-        gateway = await self.find_gateway(
-            scan_filter=self.connection_config.scan_filter,  # secure disabled by default
+        async for gateway in GatewayScanner(
+            self.xknx,
             local_ip=self.connection_config.local_ip,
-        )
-        self._gateway_info = gateway
-
-        if gateway.supports_tunnelling_tcp:
-            await self._start_tunnelling_tcp(
-                gateway_ip=gateway.ip_addr,
-                gateway_port=gateway.port,
-            )
-        elif gateway.supports_tunnelling:
-            await self._start_tunnelling_udp(
-                gateway_ip=gateway.ip_addr,
-                gateway_port=gateway.port,
-            )
-        elif gateway.supports_routing:
-            await self._start_routing(local_ip=self.connection_config.local_ip)
+            scan_filter=self.connection_config.scan_filter,  # secure disabled by default
+        ).async_scan():
+            self._gateway_info = gateway
+            try:
+                if gateway.supports_tunnelling_tcp:
```

Comment:
We should use `self._gateway_info` here since its set one line above or remove it if we dont need it (guess we have some logs where its used but havent looked)

Reply:
It's only used in HA diagnostics. We could move it to the break statement.